books/free-programming-books-subjects.md

```diff
@@ -659,6 +659,7 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Software Architecture
 
+* [A Primer on Design Patterns](https://leanpub.com/aprimerondesignpatterns) - Rahul Batra (HTML)
```

Comment:
- As it can be see in CONTRIBUTING or other similar books on this platform, add notes for Leanpub
- also complete resource formats 😉

```markdown
* [A Primer on Design Patterns](https://leanpub.com/aprimerondesignpatterns) - Rahul Batra (HTML, PDF, EPUB, Kindle) *(Leanpub account or valid email requested)*
```

Reply:
Ah thanks, missed that, will add it in. This book didn't _really_ require a Leanpub account nor valid email, it's directly accessible at https://leanpub.com/aprimerondesignpatterns/read, but I see that the book's landing page doesn't show that. I'll link the HTML format to https://leanpub.com/aprimerondesignpatterns/read, is there a cleaner/better way?